eth/vm/state.py

```diff
@@ -299,7 +316,7 @@ def __init__(self, vm_state: BaseState) -> None:
     def __call__(self, transaction: BaseOrSpoofTransaction) -> 'BaseComputation':
         valid_transaction = self.validate_transaction(transaction)
         message = self.build_evm_message(valid_transaction)
-        computation = self.build_computation(message, valid_transaction)
+        computation = self.build_computation(message, valid_transaction, self.vm_state.tracer)
```

Comment:
Here's somewhere it'd be difficult to thread the tracer too, this might be the origin of the context manager? I've exhausted my timebox for looking into this PR so I don't have an answer but I expect that getting a tracer here isn't insurmountable.

Looking at this now, BaseState is long-lived but Message isn't, maybe it's a better place to attach the tracer?